packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -0,0 +1,470 @@
+      });
+
+      expect(handleChange.callCount).to.equal(2);
+      expect(handleChange.args[0][1]).to.equal(80);
+      expect(handleChange.args[1][1]).to.equal(78);
+    });
+  });
+
+  describe('range', () => {
+    it('should support keyboard', () => {
+      const { container } = render(<Slider defaultValue={[20, 30]} />);
+      const thumb1 = container.querySelectorAll('[role="slider"]')[0];
+      const thumb2 = container.querySelectorAll('[role="slider"]')[1];
+
+      fireEvent.keyDown(thumb1, {
+        key: 'ArrowRight',
+      });
+      expect(thumb1.getAttribute('aria-valuenow')).to.equal('21');
+
+      fireEvent.keyDown(thumb2, {
+        key: 'ArrowLeft',
+      });
+      expect(thumb2.getAttribute('aria-valuenow')).to.equal('29');
+    });
+
+    it('should support mouse events', () => {
+      const handleChange = spy();
+      const { container } = render(<Slider defaultValue={[20, 30]} onChange={handleChange} />);
+      stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
+        width: 100,
+        height: 10,
+        bottom: 10,
+        left: 0,
+      }));
+
+      const event = fireBodyMouseEvent('touchstart', {
+        changedTouches: touchList([{ identifier: 1, pageX: 21, pageY: 0 }]),
+      });
+      container.firstChild.dispatchEvent(event);
+
+      fireBodyMouseEvent('touchmove', {
+        changedTouches: touchList([{ identifier: 1, pageX: 22, pageY: 0 }]),
+      });
+      fireBodyMouseEvent('touchmove', {
+        changedTouches: touchList([{ identifier: 1, pageX: 22, pageY: 0 }]),
+      });
+
+      expect(handleChange.callCount).to.equal(3);
+      expect(handleChange.args[0][1]).to.deep.equal([21, 30]);
+      expect(handleChange.args[1][1]).to.deep.equal([22, 30]);
+      expect(handleChange.args[2][1]).to.equal(handleChange.args[1][1]);
+    });
+  });
+
+  describe('prop: step', () => {
+    it('should handle a null step', () => {
+      const { getByRole, container } = render(
+        <Slider
+          step={null}
+          marks={[{ value: 0 }, { value: 20 }, { value: 30 }]}
+          defaultValue={0}
+        />,
+      );
+      stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
+        width: 100,
+        height: 10,
+        bottom: 10,
+        left: 0,
+      }));
+      const thumb = getByRole('slider');
+
+      const event = fireBodyMouseEvent('touchstart', {
+        changedTouches: touchList([{ identifier: 1, pageX: 21, pageY: 0 }]),
+      });
+      container.firstChild.dispatchEvent(event);
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('20');
+
+      fireEvent.keyDown(thumb, {
+        key: 'ArrowUp',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('30');
+
+      fireEvent.keyDown(thumb, {
+        key: 'ArrowDown',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('20');
+    });
+  });
+
+  describe('prop: disabled', () => {
```

Comment:
Should it also test for aria-disabled? What about focusability?

Reply:
Do we need to apply aira-disabled? The component is not focusable when disabled.

Reply:
We don't need any aria attributes so, no we don't need it. 

> The component is not focusable when disabled.

Yeah, do we have a test for it?

There's an argument to be had if disabled elements should be focusable or not because it effectively hides them. Some might prefer to force a tabIndex and announcing this as disabled. If you don't want to support this use case we don't need a test for aria-disabled.

Reply:
I have added a tab index logic test to make sure the element is not focusable, following your first comment.